core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -576,15 +580,22 @@ public List<Task> getPendingTasksForTaskType(String taskType) {
     }
 
     public boolean addEventExecution(EventExecution eventExecution) {
-        return executionDAOFacade.addEventExecution(eventExecution);
+        if (isEventExecutionPersistenceEnabled) {
+            return executionDAOFacade.addEventExecution(eventExecution);
+        }
+        return true;
     }
 
     public void removeEventExecution(EventExecution eventExecution) {
-        executionDAOFacade.removeEventExecution(eventExecution);
+        if (isEventExecutionPersistenceEnabled) {
+            executionDAOFacade.removeEventExecution(eventExecution);
+        }
     }
 
     public void updateEventExecution(EventExecution eventExecution) {
-        executionDAOFacade.updateEventExecution(eventExecution);
+        if (isEventExecutionPersistenceEnabled) {
```

Comment:
Why are we only updating event execution when the eventExecutionPersistenceEnabled is set to true. We should update in any case. Please help me understand.